public/app/features/logs/components/LogDetailsRow.tsx

```diff
@@ -190,11 +195,46 @@ class UnThemedLogDetailsRow extends PureComponent<Props, State> {
     });
   }
 
+  generateMultiVal(value: string[], showCopy?: boolean) {
+    const { theme } = this.props;
+    const styles = getStyles(theme);
+
+    return (
+      <table>
+        <tbody>
+          {value.map((val, i) => {
+            return (
+              <tr key={`${val}-${i}`}>
+                <td>
+                  {val}
+                  {showCopy && val !== '' && (
+                    <div className={cx('show-on-hover', styles.copyButton)}>
+                      <ClipboardButton
```

Comment:
It's minor feedback but this block is duplicated. It's related to my previous comment: I was wondering if it'd make sense to always use `generateMultiVal` even for a single value to simplify the code or it creates styling issues?

Reply:
The behavior is really different between one value and multiple values, simply showing text vs building a table, so I would rather keep them separate. That said, you're right about the duplicated ClipboardButton code so I moved that to its own function. Hopefully that is an okay compromise.